xdsl/interpreters/transform.py

```diff
@@ -22,8 +37,27 @@ def run_named_sequence_op(
     ) -> PythonValues:
         return interpreter.run_ssacfg_region(op.body, args, op.sym_name.data)
 
+    @impl(transform.ApplyRegisteredPassOp)
+    def run_apply_registered_pass_op(
+        self,
+        interpreter: Interpreter,
+        op: transform.ApplyRegisteredPassOp,
+        args: PythonValues,
+    ) -> PythonValues:
+        pass_name = op.pass_name.data
+        requested_by_user = PipelinePass.build_pipeline_tuples(
+            self.passes, parse_pipeline.parse_pipeline(pass_name)
+        )
```

Comment:
I just realized we don't need this whole dance of tuples etc any more since passes are now immutable. I'll open a quick PR hopefully making this part of xDSL a tiny bit less weird

Reply:
#4441

Reply:
But also iterables as local variables scare me a little bit, as attempting to introspect them while debugging will alter the execution of the program, I would probably inline this into the call below